test/tools/PSxUnit/Program.cs

```diff
@@ -0,0 +1,226 @@
+
+            foreach (var assemblyName in dependencyModel.GetRuntimeAssemblyNames(RuntimeEnvironment.GetRuntimeIdentifier()))
+            {
+                try
+                {
+                    var assembly = Assembly.Load(assemblyName);
+                    foreach (var type in assembly.DefinedTypes)
+                    {
+#pragma warning disable CS0618
+                        if (type == null || type.IsAbstract || type == typeof(DefaultRunnerReporter).GetTypeInfo() || type == typeof(DefaultRunnerReporterWithTypes).GetTypeInfo() || type.ImplementedInterfaces.All(i => i != typeof(IRunnerReporter)))
+                            continue;
+#pragma warning restore CS0618
+                    
+                        var ctor = type.DeclaredConstructors.FirstOrDefault(c => c.GetParameters().Length == 0);
+                        if (ctor == null)
+                        {
+                            Console.ForegroundColor = ConsoleColor.Yellow;
+                            Console.WriteLine($"Type {type.FullName} in assembly {assembly} appears to be a runner reporter, but does not have an empty constructor.");
+                            Console.ResetColor();
+                            continue;
+                        }
+
+                        result.Add((IRunnerReporter)ctor.Invoke(new object[0]));
+                    }
+                }
+                catch
+                {
+                    continue;
+                }
+            }
+
+            return result;
+        }
+
+        public static void RunTests(Options o)
+        {
+            
+            var nullMessage = new Xunit.NullMessageSink();
+            var discoveryOptions = TestFrameworkOptions.ForDiscovery();
+            using (var c = new XunitFrontController(AppDomainSupport.Denied, o.Assembly, null, false))
+            {
+                var tv = new TestDiscoverySink();
+                var excludeTestCaseSet = new TestDiscoverySink();
+                c.Find(true, tv, discoveryOptions);
+                tv.Finished.WaitOne();
+                foreach (var tc in tv.TestCases)
+                {
+                    var method = tc.TestMethod.Method;
+                    var attributes = method.GetCustomAttributes(typeof(FactAttribute));
+                    foreach (ReflectionAttributeInfo at in attributes)
+                    {
+                        var result = at.GetNamedArgument<string>("Skip");
+                        if (result != null)
+                        {
+                            Console.WriteLine("SKIPPY! {0} because {1}", method, result);
+                        }
+
+                        if (o.TestType != TestType.All)
+                        {
+                            if (!at.ToString().EndsWith(o.TestType.ToString()))
+                            {
+                                excludeTestCaseSet.TestCases.Add(tc);
+                            }
+                        }
```

Comment:
Maybe this can be moved before the block which checks for "Skip". So if the testcase does not match the filter then we dont need to check if it is skipped.